spec/integration/apply_spec.rb

```diff
@@ -121,20 +123,48 @@
           expect(logs.first['message']).to eq(conn_info('ssh')[:user])
         end
 
-        it 'respects show_diff configuration' do
-          diff_string = <<~DIFF
-            @@ -1 +1 @@
-            -Silly string
-            \\ No newline at end of file
-            +Silly string (get it?)
-            \\ No newline at end of file
-          DIFF
+        context 'with show_diff configured' do
+          let(:apply_settings) { { 'show_diff' => true } }
 
-          results = run_cli_json(%w[plan run settings::show_diff -t nix_agents], project: project)
+          it 'respects show_diff configuration' do
+            diff_string = <<~DIFF
+              @@ -1 +1 @@
+              -Silly string
+              \\ No newline at end of file
+              +Silly string (get it?)
+              \\ No newline at end of file
+            DIFF
 
-          results.each do |result|
-            expect(result['status']).to eq('success')
-            expect(result['value']['report']['logs'][0]['message']).to include(diff_string)
+            results = run_cli_json(%w[plan run settings::show_diff -t nix_agents], project: project)
+
+            results.each do |result|
+              expect(result['status']).to eq('success')
+              expect(result['value']['report']['logs'][0]['message']).to include(diff_string)
+            end
+          end
+        end
+
+        context 'without trace configured' do
+          it 'does not include backtrace in an error message' do
+            results = run_cli_json(['apply', '-e', '1 < blue', '-t', 'nix_agents'], project: project)
+
+            results.each do |result|
+              expect(result['status']).to eq('failure')
+              expect(result.dig('value', '_error', 'msg').lines.count).to eq(1)
+            end
+          end
+        end
+
+        context 'with trace configured' do
+          let(:apply_settings) { { 'trace' => true } }
+
+          it 'includes backtrace in an error message' do
+            results = run_cli_json(['apply', '-e', '1 < blue', '-t', 'nix_agents'], project: project)
+
+            results.each do |result|
+              expect(result['status']).to eq('failure')
+              expect(result.dig('value', '_error', 'msg').lines.count).to be > 1
```

Comment:
Adding a match on `/evaluator/compare_operator.rb:65:in `cmp_Numeric'/` or another substring would be good so this doesn't pass if a *different* error is raised.